pkg/core/blockchain.go

```diff
@@ -1289,15 +1304,20 @@ func appendTokenTransferInfo(transferData *state.TokenTransferInfo,
 
 func (bc *Blockchain) removeOldTransfers(index uint32) time.Duration {
 	bc.log.Info("starting transfer data garbage collection", zap.Uint32("index", index))
-	start := time.Now()
-	h, err := bc.GetHeader(bc.GetHeaderHash(index))
-	if err != nil {
+	var (
+		err     error
+		kept    int64
+		removed int64
+		start   = time.Now()
+		ts, ok  = bc.gcBlockTimes.Get(index)
```

Comment:
It won't work if you shutdown the node immediately after persist. Some transfers will inevitably stay in the DB.

Reply:
Yes. It's not perfect. But the next cycle will kick them out. I've thought about moving block/header removal into GC cycle, but it can't be done easily, we can't easily iterate over something and drop blocks before N because blocks are stored by hashes.